app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java

```diff
@@ -556,4 +559,39 @@ private String searchValueDataElement(String dataElement, List<TrackedEntityData
 
         return "";
     }
+
+    @Override
+    public Single<CoordinateViewModel> getGeometryModel(String programUid, FlowableProcessor<RowAction> processor) {
+        return Single.fromCallable(() -> {
+            ArrayList<EventStatus> nonEditableStatus = new ArrayList<>();
+            nonEditableStatus.add(EventStatus.COMPLETED);
+            nonEditableStatus.add(EventStatus.SKIPPED);
+            boolean shouldBlockEdition = eventUid != null &&
+                    !d2.eventModule().eventService().blockingIsEditable(eventUid) &&
+                    nonEditableStatus.contains(d2.eventModule().events().uid(eventUid).blockingGet().status());
+            FeatureType featureType = programStageWithId(stageUid).blockingFirst().featureType();
```

Comment:
I think unit test is missing 👍🏻 . (this leads to the previous question for having an eassier test)

Reply:
added!